lib/mirage_block_partition.mli

```diff
@@ -18,6 +18,6 @@ module Make(B : Mirage_block.S) : sig
   (** [get_offset b] is the sector offset of the partition relative to the
       underlying block device. *)
 
-      val connect' : int -> int -> t
+  val connect' : int64 -> t -> t * t
```

Comment:
Now you are returning two partitions...

Reply:
I just made new changes, can you check, please

Reply:
Yes, now the code definitely does not type check. `connect'` needs to take a `B.t` as argument instead of a `t` as well.

Reply:
Thanks, I will surely look into that as well, can you provide some hints?

Reply:
Look at the functions `connect`, `subpartition` and `partition`. Try to explain what each line does. Then try to write down in words what `connect'` needs to do and what steps are involved in order to do that.

Reply:
thanks okay, Working on that....

Reply:
Yes, im trying to understand that now..... will it take the offset and length as argument?

Reply:
I think the offset is the number of sectors between the beginning of the partition and the beginning of the disk right?

Reply:
Offset is the number of sectors (offset) from the beginning of the block device to the beginning of the partition.

Reply:
okay, thanks for the clarification.